test/e2e/lib/gutenberg/gutenberg-editor-component.js

```diff
@@ -273,100 +276,129 @@ export default class GutenbergEditorComponent extends AsyncBaseContainer {
+				blockSettings = { prefix: 'coblocks-', blockClass: 'logos' };
 				break;
 			case 'Dynamic HR':
-				prefix = 'coblocks-';
-				blockClass = 'dynamic-separator';
-				break;
-			case 'Heading':
-				break;
-			case 'Layout Grid':
-				prefix = 'jetpack-';
+				blockSettings = { prefix: 'coblocks-', blockClass: 'dynamic-separator' };
 				break;
 			case 'Blog Posts':
-				prefix = 'a8c-';
+				blockSettings = { prefix: 'a8c-' };
 				break;
 			case 'Subscription Form':
-				prefix = 'jetpack-';
-				blockClass = 'subscriptions';
+				blockSettings = { prefix: 'jetpack-', blockClass: 'subscriptions' };
 				break;
+			case 'Layout Grid':
 			case 'Tiled Gallery':
-				prefix = 'jetpack-';
-				break;
 			case 'Contact Info':
-				prefix = 'jetpack-';
-				hasChildBlocks = true;
-				break;
 			case 'Slideshow':
-				prefix = 'jetpack-';
+				blockSettings = { prefix: 'jetpack-' };
 				break;
 			case 'Star Rating':
-				prefix = 'jetpack-';
-				blockClass = 'rating-star';
-				break;
-			case 'Masonry':
-				prefix = 'coblocks-';
-				blockClass = 'gallery-masonry';
+				blockSettings = { prefix: 'jetpack-', blockClass: 'rating-star' };
 				break;
 		}
 
-		const selectorAriaLabel = ariaLabel || `Block: ${ title }`;
+		return { ...defaultSettings, ...blockSettings };
+	}
+	// return blockID - top level block id which is looks like `block-b91ce479-fb2d-45b7-ad92-22ae7a58cf04`. Should be used for further interaction with added block.
+	async addBlock( title ) {
+		const { ariaLabel, prefix, blockClass, initsWithChildFocus } = this.getBlockSelectorSettings(
+			title
+		);
 
+		// TODO Remove the `deprecatedInserterBlockItemSelector` definition and usage after we activate GB 9.4.0 on production.
+		const deprecatedInserterBlockItemSelector = `.edit-post-layout__inserter-panel .block-editor-inserter__block-list button.editor-block-list-item-${ prefix }${ blockClass }`;
 		const inserterBlockItemSelector = By.css(
-			`.edit-post-layout__inserter-panel .block-editor-inserter__block-list button.editor-block-list-item-${ prefix }${ blockClass }`
+			`.edit-post-layout__inserter-panel .block-editor-block-types-list button.editor-block-list-item-${ prefix }${ blockClass }, ${ deprecatedInserterBlockItemSelector }`
 		);
 
 		let insertedBlockSelector = By.css(
 			`.block-editor-block-list__block.${
-				hasChildBlocks ? 'has-child-selected' : 'is-selected'
-			}[aria-label*='${ selectorAriaLabel }']`
+				initsWithChildFocus ? 'has-child-selected' : 'is-selected'
+			}[aria-label*='${ ariaLabel }']`
 		);
 
 		// @TODO: Remove this condition when Gutenberg v9.1 is deployed for all sites.
 		if ( title === 'Heading' ) {
-			const deprecatedSelector = insertedBlockSelector.value.replace(
-				selectorAriaLabel,
-				'Write heading…'
-			);
+			const deprecatedSelector = insertedBlockSelector.value.replace( ariaLabel, 'Write heading…' );
 			insertedBlockSelector = By.css( `${ insertedBlockSelector.value }, ${ deprecatedSelector }` );
 		}
```

Comment:
Seems like we can remove this altogether?

See https://github.com/Automattic/wp-calypso/pull/47533/files#diff-b7f8735b885b065dc18c161fbf8de3c5bfecafd5fd04319842a8eea5f5c32180L364

Reply:
We can remove the whole block of code. The deprecated selector was only needed for GB < 9.1. The condition there is to make sure it applies only when inserting the `Heading` block Thanks for pointing this out :)